File: Weather.Business/V1/Logic/Aspnet/User/UserRole/UserRoleModel.cs
using System;
using System.Collections.Generic;
using System.Text;
using Weather.Data.V1;

namespace Weather.Business.V1
{
    public class UserRoleFilterModel : BaseQueryFilterModel
    {
        public Guid? RoleId { get; set; }
    }
    public class UserRoleCreateRequestModel
    {
        public string RoleName { get; set; }
        public string Description { get; set; }
        public bool EnableDelete { get; set; } // Lock role
        public Guid CreatedByUserId { get; set; }
        //public DateTime CreatedOnDate { get; set; }
        public List<Guid> RightList { get; set; }
    }
    public class UserRoleUpdateRequestModel
    {
        public Guid RoleId { get; set; }

        public string RoleName { get; set; }
        public string Description { get; set; }
        //public bool EnableDelete { get; set; } // Lock role

        public Guid LastModifiedByUserId { get; set; }
        //public DateTime LastModifiedOnDate { get; set; }
        public List<Guid> RightList { get; set; }
    }
    public class UserRoleDeleteResponseModel : BaseDeleteResponseModel
    {

    }
}

File: _VienKhiTuongThuyVan/KTTVPortal/src/app/CMS/services/roles/user-role.model.ts
import { BaseQueryFilter } from 'src/app/_layout/utils/common-classes';

export class UserRole {
    public RoleId: string;
    public RoleName: string;
    public LoweredRoleName: string;
    public Description: string;
    public EnableDelete: boolean;
    public CreatedByUserId: string;
    public CreatedOnDate: string;
    public LastModifiedByUserId: string;
    public LastModifiedOnDate: string;
    public IdmRightsInRole: Array<any>;
    public RightList: Array<string>;
}

export class UserRoleFilter extends BaseQueryFilter {
    public RoleId: string = null;
}

export class UserRoleCreateRequestModel {
    public RoleName: string;
    public Description: string;
    public CreatedByUserId: string;
    public RightList: Array<string>;
}
export class UserRoleUpdateRequestModel {
    public RoleId: string;
    public RoleName: string;
    public Description: string;
    public LastModifiedByUserId: string;
    public RightList: Array<string>;
}
